api/server/router/container/exec.go

```diff
@@ -154,3 +156,31 @@ func (s *containerRouter) postContainerExecResize(ctx context.Context, w http.Re
 
 	return s.backend.ContainerExecResize(vars["name"], height, width)
 }
+
+func (s *containerRouter) postContainerExecKill(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	version := httputils.VersionFromContext(ctx)
+	if versions.LessThan(version, "1.42") {
+		return errdefs.InvalidParameter(fmt.Errorf("postContainerExecKill requires API version 1.42, but the Docker daemon API version is %s", version))
+	}
+	if err := httputils.ParseForm(r); err != nil {
+		return errdefs.InvalidParameter(err)
+	}
+
+	var sig syscall.Signal
+	name := vars["name"]
+
+	// If we have a signal, look at it. Otherwise, do nothing
+	if sigStr := r.Form.Get("signal"); sigStr != "" {
+		var err error
+		if sig, err = signal.ParseSignal(sigStr); err != nil {
```

Comment:
Is this what we do in the container API? I think we should have this in `ContainerExecKill` rather than here.

Reply:
This block appears to be a copy-paste from the `postContainersKill`:

https://github.com/moby/moby/blob/9a6ff685a80639995311c0572d662360b55796d9/api/server/router/container/container_routes.go#L248-L254